### ISSUE TITLE
Folders: Lookup of second-level folders is case-sensitive

### ISSUE THREAD

**Issue by joseorsini** [dotCMS/core#9463]
Reproduced in 3.5. 

https://my.dotcms.com/tickets/detail.dot?id=7fdca928-07c3-4785-816e-220ac4a54293 

Steps. 
1. On demo site, go to Content Search Manager.
2. Create a new Content (Generic) Content. 
3. On the TinyMCE Editor, click on Insert/Edit image.
4. On "Source", click on the search Button.
5. Expand the demo.dotcms.com site. Folders look ordered right.
6. Expand "application" folder. 

Screenshot: 
<img width="1379" alt="screen shot 2016-07-29 at 4 24 15 pm" src="https://cloud.githubusercontent.com/assets/1717110/17262363/ede49ae8-55a8-11e6-85ff-871106307c72.png">

The query that hits the DB in order to pull folders is the following: 

```
SELECT folder.* from folder folder, inode folder_1_, identifier identifier where folder.identifier = identifier.id and folder_1_.type = 'folder' and folder_1_.inode = folder.inode and identifier.parent_path = '/application/' and identifier.host_inode = '48190c8c-42c4-46af-8d1a-0cd5db894797' order by folder.title;
                inode                 |     name      |     title     | show_on_menu | sort_order | files_masks |              identifier              |          default_file_type           |          mod_date
--------------------------------------+---------------+---------------+--------------+------------+-------------+--------------------------------------+--------------------------------------+----------------------------
 303676ba-8bca-45a2-a3e7-04e1b4ce92f2 | login         | Login         | f            |          0 |             | 2abe28a0-0bd3-47c4-aed0-b4db62625bb0 | 33888b6f-7a8e-4069-b1b6-5c1aa9d0a48d | 2014-03-21 10:11:18.793413
 3f0846a6-3260-473c-9f71-3cce1cbdd9a1 | myaccount     | My Account    | f            |          0 |             | c37cb06e-d022-4ac0-a9de-d100b4993224 | 33888b6f-7a8e-4069-b1b6-5c1aa9d0a48d | 2014-03-21 10:11:18.793413
 e3d95194-9209-426a-80d0-7b27bb608237 | ajax          | ajax          | f            |          0 |             | 3988a0ab-ecae-4583-ba82-0bd8371d9be2 | 33888b6f-7a8e-4069-b1b6-5c1aa9d0a48d | 2014-03-21 10:11:18.793413
 2efd76cd-93e1-45d3-84ff-07a650711a35 | comments      | comments      | f            |          0 |             | e1c2dead-3ea8-45e1-beaa-0e90bed4df6a | 33888b6f-7a8e-4069-b1b6-5c1aa9d0a48d | 2015-04-14 13:55:47.995
 47663ed5-b009-4ee5-9390-ef6dce783f16 | createaccount | createAccount | f            |          0 |             | 638274ff-d1c9-4e0f-996e-657d88201f85 | 33888b6f-7a8e-4069-b1b6-5c1aa9d0a48d | 2014-03-21 10:11:18.793413
 9125d787-00cd-4717-bbea-84a6660ece62 | spring        | spring        | f            |          0 |             | be731dec-372c-41b1-b8ec-894ec939d29a | 33888b6f-7a8e-4069-b1b6-5c1aa9d0a48d | 2014-03-21 10:11:18.793413
 8c31a241-9dce-434a-8b77-ec902155307d | themes        | themes        | f            |          0 |             | 4aa37d39-41c0-4a99-bb22-f7580dbe9dcc | 33888b6f-7a8e-4069-b1b6-5c1aa9d0a48d | 2014-03-21 10:11:18.793413
 d274cc74-305f-4a61-9bdf-4695a01ab035 | vtl           | vtl           | f            |          0 |             | 8b2d19e8-80eb-4184-86b3-1376674b683a | 33888b6f-7a8e-4069-b1b6-5c1aa9d0a48d | 2014-07-25 16:34:07.467
(8 rows)
```

Since case is different for some folders, folders with uppercase titles take precedence upon a DB lookup. 
## 

All lookups in our FolderAPI and factory classes should be case-insensitive, so the order is right and it becomes easier for content editors to find folders in Website Browser and in other places of the CMS.

**Comment by joseorsini** [comment]
PR: https://github.com/dotCMS/core/pull/9703

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/9898

**Comment by bryanboza** [comment]
Fixed, tested on master-3.7 // postgres // ff
